### PR TITLE
Add basic module wiring in app.go

### PR DESCRIPTION
**Pull request by mattverse**
Adds some more basic wiring for app.go

### CODE REVIEW

File: app/keepers/modules.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ import (
 	ica "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts"
 
 	_ "github.com/osmosis-labs/osmosis/v12/client/docs/statik"
+	concentrated_liquidity "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity"
```
Comment (nicolaslara) [inline app/keepers/modules.go:31]: can we use `concentratedliquidity` here? or change it to use the underscore in app/modules.go. Just unifying it

**Review by nicolaslara — APPROVED** [review]
other than my comment above, this looks good to me

**Comment by mattverse** [comment]
Converting to draft to add init genesis export genesis

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/CosmWasm/wasmd v0.28.0-osmo-v12
 	github.com/cosmos/cosmos-proto v1.0.0-alpha8
-	github.com/cosmos/cosmos-sdk v0.46.2
+	github.com/cosmos/cosmos-sdk v0.46.3
```
Comment (czarcas7ic) [inline go.mod:8]: are we good with bumping this?

File: proto/osmosis/concentrated-liquidity/genesis.proto [hunk 3]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+package osmosis.concentratedliquidity.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types";
+
+// GenesisState defines the gamm module's genesis state.
```
Comment (czarcas7ic) [inline proto/osmosis/concentrated-liquidity/genesis.proto:9]:
```suggestion
// GenesisState defines the concentrated liquidity module's genesis state.
```

File: x/concentrated-liquidity/genesis.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
+)
+
+func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
+
+}
+
+// ExportGenesis returns the capability module's exported genesis.
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/genesis.go:13]:
```suggestion
// ExportGenesis returns the concentrated liquidity module's exported genesis.
```

File: x/concentrated-liquidity/module.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+func (am AppModule) Route() sdk.Route {
+	return sdk.Route{}
+}
+
+// QuerierRoute returns the gamm module's querier route name.
+func (AppModule) QuerierRoute() string { return types.RouterKey }
+
+// LegacyQuerierHandler returns the x/gamm module's sdk.Querier.
+func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/module.go:92]:
```suggestion
// QuerierRoute returns the concentrated liquidity module's querier route name.
func (AppModule) QuerierRoute() string { return types.RouterKey }

// LegacyQuerierHandler returns the x/concentrated liquidity module's sdk.Querier.
func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
```

File: x/concentrated-liquidity/module.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+	return func(sdk.Context, []string, abci.RequestQuery) ([]byte, error) {
+		return nil, fmt.Errorf("legacy querier not supported for the x/%s module", types.ModuleName)
+	}
+}
+
+// InitGenesis performs genesis initialization for the twap module.
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/module.go:98]:
```suggestion
// InitGenesis performs genesis initialization for the concentrated liquidity module.
```

File: x/concentrated-liquidity/module.go [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+	var genState types.GenesisState
+	// Initialize global index to index in genesis state
+	cdc.MustUnmarshalJSON(gs, &genState)
+
+	am.k.InitGenesis(ctx, genState)
+
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the exported genesis state as raw bytes for the twap.
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/module.go:110]:
```suggestion
// ExportGenesis returns the exported genesis state as raw bytes for the concentrated liquidity.
```

**Review by czarcas7ic — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (mattverse) [reply go.mod:8]: This should be redundent right? We're replacing it any ways, i think its just an annoying proto thing

**Pull request merged**